src/tools/materialphysicalproperties/slags.py

```diff
@@ -127,12 +125,129 @@ def calculate(self, **state):
         x = amount_fractions(y)
         return super().calculate(T=T, x=x)
 
+class ModifiedUrbainViscosityTx(Model):
+    """
+    A model that describes the variation in the dynamic viscosity of liquid
+    slag as a function of temperature and composition expressed in mole
+    fraction. This "Modified" Urbain Model is more appropriate for Basic
+    Oxygen Steelmaking Slags than the standard, generic UrbainViscosity model.
+    """
+
+    def __init__(self):
+        state_schema = {'T': {'required': True, 'type': 'float', 'min': 0.0},
+                        'x': {'required': True, 'type': 'dict'}}
+        super().__init__('Slag', 'Dynamic Viscosity', 'mu', '\\mu', 'Pa.s',
+                         state_schema, None, None)
```

Comment:
Please pass values for the `references` and `datasets` parameters to the base class constructor for the following models in this module:

- [ ] UrbainViscosityTx
- [ ] RiboudViscosityTx
- [ ] ModifiedUrbainViscosityTx

For an example of how to do this, refer to the `auxi.tools.materialphysicalproperties.core.Model` class docstring.